File: trail2/두 가지로 열리는 자물쇠/a-two-way-lock.py
N = int(input())
a1, b1, c1 = map(int, input().split())
a2, b2, c2 = map(int, input().split())

# Please write your code here.
ans = 0

def f(x, y):
    return (N - abs(x - y)) <= 2 or abs(x - y) <= 2

for i in range(1, N+1):
    for j in range(1, N+1):
        for k in range(1,N+1):

            a = f(i, a1) and f(j, b1) and f(k, c1)
            b = f(i, a2) and f(j, b2) and f(k, c2)

            if a or b:
                ans += 1

print(ans)

File: trail2/두 가지로 열리는 자물쇠/a-two-way-lock.java
import java.util.Scanner;

public class Main {
    
    static int n;

    public static boolean f(int x, int y) {
        return Math.abs(x-y) <= 2 || (n - Math.abs(x-y)) <= 2;
    }
        
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        n = sc.nextInt();
        int a = sc.nextInt();
        int b = sc.nextInt();
        int c = sc.nextInt();
        int a2 = sc.nextInt();
        int b2 = sc.nextInt();
        int c2 = sc.nextInt();
        // Please write your code here.

    
        int ans = 0;

        for (int i = 1; i < n+1; i++) {
            for (int j = 1; j < n+1; j++) {
                for (int k = 1; k < n+1; k++) {

                    boolean x = f(i, a) && f(j, b) && f(k, c);
                    boolean y = f(i, a2) && f(j, b2) && f(k, c2);

                    if (x || y) {
                        ans++;
                    }
                }
            }
        }
        System.out.println(ans);
    }
}